magda-storage-api/src/MagdaMinioClient.ts

```diff
@@ -17,26 +26,41 @@ export default class MagdaMinioClient implements ObjectStoreClient {
             secretKey,
             region
         });
-        this.client.makeBucket(this.bucketName, region, (err: Error) => {
-            if (err) {
-                if (
-                    err.message ===
-                    "Your previous request to create the named bucket succeeded and you already own it."
-                ) {
-                    return console.log(
-                        "Bucket " + this.bucketName + " already exists 👍"
-                    );
-                } else {
-                    return console.log("😢 Error creating bucket: ", err);
+        this.region = region;
+        this.createBucket(this.bucketName);
+    }
+
+    createBucket(bucket: string): Promise<CreateBucketResponse> {
+        return new Promise((resolve, reject) => {
+            return this.client.makeBucket(bucket, this.region, (err: Error) => {
+                if (err) {
+                    if (
+                        err.message ===
+                        "Your previous request to create the named bucket succeeded and you already own it."
+                    ) {
+                        return resolve({
+                            message: "Bucket " + bucket + " already exists 👍",
+                            success: false
+                        });
+                    } else {
+                        console.log("😢 Error creating bucket: ", err);
+                        return reject({
```

Comment:
I don't think it's technically incorrect, but it's a lot more usual to [reject with an error](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/reject) if that's ok :)
```suggestion
                        return reject(err);
```